### ISSUE TITLE
io.quarkus:quarkus-maven-plugin error on ppcle and s390x linux

### ISSUE THREAD

**Issue by LongyuZhang** [quarkus-qe/quarkus-openshift-test-suite#82]
Hi there,
I am working on running quarkus-openshift test on x84, ppc64le, and s390x linux machines.
The x84 works fine to pass most tests, but for ppc64le and s390x, all tests (e.g. HttpOpenShiftIT) with ```io.quarkus:quarkus-maven-plugin``` failed with error message ```Unknown maven version output format. Expected at least one line!```.
Some combinations I tried with running results are: 
- host machine x86, Openshift Platform 4.5 on Intel/ppcle/s390x, Passed
- host machine ppcle, Openshift Platform 4.3/4.5 on ppcle/intel, failed with ```io.quarkus:quarkus-maven-plugin```
- host machine s390x, Openshift Platform 4.5 on s390x/intel, failed with ```io.quarkus:quarkus-maven-plugin```

More detailed error message is:
```
16:23:23  [ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.6.0.Final:build (default) on project security-https-1way: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
16:23:23  [ERROR] 	[error]: Build step io.quarkus.kubernetes.deployment.KubernetesProcessor#build threw an exception: java.lang.IllegalStateException: Unknown maven version output format. Expected at least one line!
16:23:23  [ERROR] 	at io.dekorate.utils.Maven.getVersionFromOutput(Maven.java:72)
16:23:23  [ERROR] 	at io.dekorate.utils.Maven.getVersion(Maven.java:58)
16:23:23  [ERROR] 	at io.dekorate.project.MavenInfoReader.getInfo(MavenInfoReader.java:73)
16:23:23  [ERROR] 	at io.dekorate.project.FileProjectFactory.lambda$getProjectInfo$2(FileProjectFactory.java:78)
16:23:23  [ERROR] 	at io.dekorate.project.FileProjectFactory$$Lambda$972/0000000000000000.apply(Unknown Source)
16:23:23  [ERROR] 	at java.base/java.util.Optional.map(Optional.java:265)
16:23:23  [ERROR] 	at io.dekorate.project.FileProjectFactory.getProjectInfo(FileProjectFactory.java:78)
16:23:23  [ERROR] 	at io.dekorate.project.FileProjectFactory.createInternal(FileProjectFactory.java:52)
16:23:23  [ERROR] 	at io.dekorate.project.FileProjectFactory.create(FileProjectFactory.java:41)
16:23:23  [ERROR] 	at io.quarkus.kubernetes.deployment.KubernetesProcessor.createProject(KubernetesProcessor.java:738)
16:23:23  [ERROR] 	at io.quarkus.kubernetes.deployment.KubernetesProcessor.build(KubernetesProcessor.java:252)
16:23:23  [ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
16:23:23  [ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
16:23:23  [ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
16:23:23  [ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
16:23:23  [ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:932)
16:23:23  [ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
16:23:23  [ERROR] 	at io.quarkus.builder.BuildContext$$Lambda$453/0000000000000000.run(Unknown Source)
16:23:23  [ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
16:23:23  [ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
16:23:23  [ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
16:23:23  [ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
16:23:23  [ERROR] 	at java.base/java.lang.Thread.run(Thread.java:836)
16:23:23  [ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
```
and when I use ```./mvnw -v``` to check the most machine mvn verion, I got:
```18:14:35  [1mApache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T15:00:29-04:00)[m
18:14:35  Maven home: /home/jenkins/.m2/wrapper/dists/apache-maven-3.6.1-bin/38pn40mp89t5c94bjdbeod370m/apache-maven-3.6.1
18:14:35  Java version: 11.0.8, vendor: AdoptOpenJDK, runtime: /home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image
18:14:35  Default locale: en_US, platform encoding: UTF-8
18:14:35  OS name: "linux", version: "3.10.0-693.17.1.el7.ppc64le", arch: "ppc64le", family: "unix"
```
and
```10:15:58  Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T12:00:29-07:00)
10:15:58  Maven home: /home/jenkins/.m2/wrapper/dists/apache-maven-3.6.1-bin/38pn40mp89t5c94bjdbeod370m/apache-maven-3.6.1
10:15:58  Java version: 11.0.8, vendor: AdoptOpenJDK, runtime: /home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image
10:15:58  Default locale: en_US, platform encoding: ANSI_X3.4-1968
10:15:58  OS name: "linux", version: "4.4.0-179-generic", arch: "s390x", family: "unix"
```

Really appreciate for suggestions to solve this problem. Thanks!

**Comment by rsvoboda** [comment]
Sounds like a bug in Quarkus, but first let me suggest few things

 - use Quarkus 1.7.1.Final instead of 1.6.0.Final
 - use Maven 3.6.2+ as suggested in https://quarkus.io/get-started/

**Comment by Ladicek** [comment]
Dekorate expects that `mvn -version` (or `mvnw -version`) output contains a line that starts with `Apache Maven`. Here's what I have, for example:

```bash
lthon@argondie:~$ mvn -version
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/lthon/software/apache-maven
Java version: 1.8.0_265, vendor: Private Build, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: cs_CZ, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-112-generic", arch: "amd64", family: "unix"
```

Compared to your output, I have one more line, the first. I wonder where your Maven comes from? Apparently you're using Maven Wrapper, so I tried this:

```bash
lthon@argondie:~/tmp$ mvn -N io.takari:maven:0.7.7:wrapper
...
lthon@argondie:~/tmp$ ./mvnw -v
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/lthon/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 1.8.0_265, vendor: Private Build, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: cs_CZ, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-112-generic", arch: "amd64", family: "unix"
```

So the version output from my Maven Wrapper looks fine, but yours does not. That's weird. It seems you're using 3.6.1, can you try updating to 3.6.3?

**Comment by LongyuZhang** [comment]
@rsvoboda @Ladicek Thanks for the help! 
I have tried to use mvn version 3.6.3 and quarkus version 1.7.1.Final, but got the same problem. The ```mvn -v``` does have the ```Apache Maven``` line, but has some wired character before it, not sure if this is the cause of the problem:
```
10:14:27  [1mApache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)[m
10:14:27  Maven home: /usr/local/apache-maven-3.6.3
10:14:27  Java version: 11.0.9, vendor: AdoptOpenJDK, runtime: /home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image
10:14:27  Default locale: en_US, platform encoding: ANSI_X3.4-1968
10:14:27  OS name: "linux", version: "4.4.0-179-generic", arch: "s390x", family: "unix"
```

**Comment by Ladicek** [comment]
Hmm, actually, if I remember correctly from today's morning when I looked at it, Dekorate expects the line to _start with_ `Apache Maven`. You not only have the ANSI escape sequence there, but also a time stamp. I wonder why that happens. I mean, it's Maven printing those lines... does it have a different, non-default logging configuration?

**Comment by Ladicek** [comment]
(And of course Dekorate depending on `mvn -v` output is really brittle. They shouldn't do that. And the only usecase they have for it is automatically figuring out a build image for Tekton. Pfff...)

**Comment by Ladicek** [comment]
Filed https://github.com/dekorateio/dekorate/issues/610 for this, too.

**Comment by LongyuZhang** [comment]
@Ladicek The time stamp is because we are integrating quarkus-openshift-test into AdoptOpenJDK test framework, and our Jenkins machines automatically add timestamp before each line. Thanks for finding the cause and opening the issue!

**Comment by Ladicek** [comment]
Ah so those prints were from a Jenkins job console? Then the time stamps were not present in the `mvn -v` output and the problem must be elsewhere.

**Comment by LongyuZhang** [comment]
Tested on both x86 and s390x machines, the s390x machine does still have extra character before ```Apache Maven```, so probably this is the reason why x86 works but not s390x and ppcle.

x86:
```
+ mvn -v
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /usr/local/apache-maven-3.6.3
Java version: 1.8.0_252, vendor: Private Build, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.4.0-179-generic", arch: "amd64", family: "unix"
```

s390x:
```
+ mvn -v
OpenJDK 64-Bit Zero VM warning: You have loaded library /usr/local/apache-maven-3.6.3/lib/jansi-native/linux64/libjansi.so which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
[1mApache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)[m
Maven home: /usr/local/apache-maven-3.6.3
Java version: 1.8.0_212, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk-s390x/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "4.4.0-179-generic", arch: "s390x", family: "unix"
```

**Comment by Ladicek** [comment]
I wonder if it's correct that Maven uses a `linux64` variant of `libjansi` on s390x. Is that a Linux LPAR or something?

In any case, can you try `mvn -B -v`? `-B` should disable colored output. If that works, perhaps Dekorate could use it as well (in addition to not throwing exceptions...)

**Comment by Ladicek** [comment]
Actually now that I'm thinking about it, Dekorate runs `mvn -v` itself, in which case Jansi should detect that its stdout is not a terminal and not use color codes at all. So it should work... Could you please contact me privately to discuss how we could run some debug code right on the machine where it's failing? Thanks!

**Comment by LongyuZhang** [comment]
@Ladicek I tried ```mvn -B -v```, it gives the same output:
```
+ mvn -B -v
OpenJDK 64-Bit Zero VM warning: You have loaded library /usr/local/apache-maven-3.6.3/lib/jansi-native/linux64/libjansi.so which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
[1mApache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)[m
Maven home: /usr/local/apache-maven-3.6.3
Java version: 1.8.0_212, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk-s390x/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "4.4.0-179-generic", arch: "s390x", family: "unix"
```
I have sent you an email so we can discuss and run some more debug code, Thanks!

**Comment by rsvoboda** [comment]
@LongyuZhang can you try dekorate master bits, https://github.com/dekorateio/dekorate/pull/611 was merged 14 hours ago

**Comment by LongyuZhang** [comment]
Thanks for the quick fix! I tried to run Quarkus 1.7.2.Final, which merged this fix, but it is not available yet in https://repo.maven.apache.org/maven2/io/quarkus/quarkus-universe-bom/ with error message:
```[ERROR] Non-resolvable import POM: Could not find artifact io.quarkus:quarkus-universe-bom:pom:1.7.2.Final in central (https://repo.maven.apache.org/maven2) @ line 74, column 25```
Will wait for it to be available to try more. Thanks.

**Comment by LongyuZhang** [comment]
The error has been fixed when I run with Quarkus 1.7.2.Final, feel free to close this issue. Thanks a lot!

**Comment by rsvoboda** [comment]
Thanks for checking 1.7.2.Final, closing.
